### PR TITLE
iPhone X support

### PR DESCRIPTION
**Pull request by bg-stripe**
r? @bdorfman-stripe 

- Added a UIView category method that returns bounds adjusted to `safeAreaInsets`
- Updated cells and `STPPaymentCardTextField` to use adjusted bounds
- Updated cells to set `backgroundColor`, not `contentView.backgroundColor`. I think it looks better when our cells are full width on iPhone X (contentView is inset to the safe area).
- Updated the example app to use `safeAreaInsets`

simulator screen shot - iphone x - 2017-12-12 at 17 57 20

simulator screen shot - iphone x - 2017-12-12 at 18 00 14

### CODE REVIEW

File: Stripe/STPAddCardViewController.m [hunk 1]
```diff
@@ -408,8 +408,7 @@ - (UITableViewCell *)tableView:(__unused UITableView *)tableView
         default:
             return [UITableViewCell new]; // won't be called; exists to make the static analyzer happy
     }
-    cell.backgroundColor = [UIColor clearColor];
-    cell.contentView.backgroundColor = self.theme.secondaryBackgroundColor;
+    cell.backgroundColor = self.theme.secondaryBackgroundColor;
```
Comment (bdorfman-stripe) [inline Stripe/STPAddCardViewController.m]: If you don't set contentView background is it clear? Do you need to set both cell background and content view background to the theme color?

File: Stripe/UIView+Stripe_SafeAreaBounds.m [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation UIView (Stripe_SafeAreaBounds)
+
+- (CGRect)boundsWithHorizontalSafeAreaInsets {
+    if (@available(iOS 11.0, *)) {
+        UIView *view = self;
+        UIEdgeInsets insets = UIEdgeInsetsZero;
+        while (view.superview != nil || UIEdgeInsetsEqualToEdgeInsets(insets, UIEdgeInsetsZero)) {
+            view = view.superview;
```
Comment (bdorfman-stripe) [inline Stripe/UIView+Stripe_SafeAreaBounds.m]: This seems like not how `safeAreaInsets` is meant to be used. Is there some other reason the insets aren't being propagated to subviews like described in the docs?

File: Stripe/STPAddCardViewController.m (hunk 1 above)
Comment (bg-stripe) [reply Stripe/STPAddCardViewController.m]: It seems like `contentView.backgroundColor` defaults to clear, but probably best to set it explicitly just it's different on other iOS versions. I'll update.

File: Stripe/UIView+Stripe_SafeAreaBounds.m (hunk 2 above)
Comment (bg-stripe) [reply Stripe/UIView+Stripe_SafeAreaBounds.m]: Ah whoops, yeah this isn't necessary – I was confused by the docs at first, and thought it didn't propagate.

**Comment by bg-stripe** [comment]
updated – re-r? @bdorfman-stripe

File: Stripe/UIView+Stripe_SafeAreaBounds.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+//
+
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface UIView (Stripe_SafeAreaBounds)
+
+/**
+ Finds the first non-zero `safeAreaInsets` in the view hierarchy, and returns
```
Comment (bdorfman-stripe) [inline Stripe/UIView+Stripe_SafeAreaBounds.h]: This comment is wrong now

Comment (bg-stripe) [reply Stripe/UIView+Stripe_SafeAreaBounds.h]: ah yep, will fix

**Comment by bdorfman-stripe** [comment]
Also maybe add a note to the changelog that this was fixed?

**Comment by bg-stripe** [comment]
👍  updated the comment for that method and added a CHANGELOG entry.

I don't think we need to worry about `viewSafeAreaInsetsDidChange` – whenever orientation changes we also get a layoutSubviews call.

**Pull request merged**